common/ast/src/parser/ast/expression.rs

```diff
@@ -100,22 +100,20 @@ pub enum Expr {
 
 #[derive(Debug, Clone, PartialEq)]
 pub enum TypeName {
-    Char(Option<u64>),
-    Varchar(Option<u64>),
-    Decimal(Option<u64>, Option<u64>),
-    Float(Option<u64>),
-    Int(Option<u64>),
-    TinyInt(Option<u64>),
-    SmallInt(Option<u64>),
-    BigInt(Option<u64>),
-    Real,
-    Double,
     Boolean,
+    TinyInt { unsigned: bool },
+    SmallInt { unsigned: bool },
+    Int { unsigned: bool },
+    BigInt { unsigned: bool },
+    Float,
+    Double,
     Date,
-    Time,
+    DateTime,
     Timestamp,
-    Interval,
-    Text,
+    Varchar,
+    Array,
```

Comment:
We decide to have `Array(TypeName)`. It's more generic.

Reply:
What is the grammar like for the type generic?

Reply:
That's great. By the way, shall we have some structural types like https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#struct_type

Reply:
Seem postgresql chose a [different syntax](https://www.postgresql.org/docs/9.1/arrays.html#:~:text=CREATE%20TABLE%20tictactoe%20(%0A%20%20%20%20squares%20%20%20integer%5B3%5D%5B3%5D%0A)%3B)

Reply:
Currently, structural type is replaced by Variant...

Reply:
> What is the grammar like for the type generic?

```
Array(tinyint)
Array(Varchar)
Array(int unsigned)
```

Reply:
addressed